packages/malloy-render/src/component/apply-renderer.tsx

```diff
@@ -24,88 +13,38 @@ import MalloyTable from './table/table';
 import {renderList} from './render-list';
 import {renderImage} from './render-image';
 import {Dashboard} from './dashboard/dashboard';
-import {LegacyChart} from './legacy-charts/legacy_chart';
 import {renderTime} from './render-time';
-
-export type RendererProps = {
-  field: Field;
-  dataColumn: DataColumn;
-  resultMetadata: RenderResultMetadata;
-  tag: Tag;
-  customProps?: Record<string, Record<string, unknown>>;
-};
-
-const RENDER_TAG_LIST = [
-  'link',
-  'image',
-  'cell',
-  'list',
-  'list_detail',
-  'bar_chart',
-  'line_chart',
-  'dashboard',
-  'scatter_chart',
-  'shape_map',
-  'segment_map',
-];
-
-const CHART_TAG_LIST = ['bar_chart', 'line_chart'];
-
-export function shouldRenderChartAs(tag: Tag) {
-  const tagNamesInOrder = Object.keys(tag.properties ?? {}).reverse();
-  return tagNamesInOrder.find(name => CHART_TAG_LIST.includes(name));
-}
-
-export function shouldRenderAs(f: Field | Explore, tagOverride?: Tag) {
-  const tag = tagOverride ?? f.tagParse().tag;
-  const tagNamesInOrder = Object.keys(tag.properties ?? {}).reverse();
-  for (const tagName of tagNamesInOrder) {
-    if (RENDER_TAG_LIST.includes(tagName)) {
-      if (['list', 'list_detail'].includes(tagName)) return 'list';
-      if (['bar_chart', 'line_chart'].includes(tagName)) return 'chart';
-      return tagName;
-    }
-  }
-
-  if (!f.isExplore() && f.isAtomicField()) return 'cell';
-  return 'table';
-}
-
-export const NULL_SYMBOL = '∅';
+import {LegacyChart} from './legacy-charts/legacy_chart';
+import {NULL_SYMBOL} from './util';
+import {RendererProps} from './types';
 
 export function applyRenderer(props: RendererProps) {
-  const {field, dataColumn, resultMetadata, customProps = {}} = props;
-  const renderAs = resultMetadata.field(field).renderAs;
+  const {dataColumn, customProps = {}} = props;
+  const field = props.dataColumn.field;
+  const renderAs = field.renderAs;
   let renderValue: JSXElement = '';
   const propsToPass = customProps[renderAs] || {};
   if (dataColumn.isNull()) {
     renderValue = NULL_SYMBOL;
   } else {
     switch (renderAs) {
       case 'cell': {
-        const resultCellValue = dataColumn.value;
-        if (valueIsNumber(field, resultCellValue)) {
+        if (dataColumn.isNumber()) {
           // TS doesn't support typeguards for multiple parameters, so unfortunately have to assert AtomicField here. https://github.com/microsoft/TypeScript/issues/26916
-          renderValue = renderNumericField(
-            field as AtomicField,
-            resultCellValue
-          );
-        } else if (valueIsString(field, resultCellValue)) {
-          renderValue = resultCellValue;
-        } else if (
-          field.isAtomicField() &&
-          (field.isDate() || field.isTimestamp())
-        ) {
+          renderValue = renderNumericField(field, dataColumn.value);
```

Comment:
Maybe should just be `renderNumericField(dataColumn)`. Need to check how it's used elsewhere...

Reply:
Ah yeah it's being used to render axis values, not just actual cells.